src/test/java/sysc4806/graduateAdmissions/controller/InterestControllerTest.java

```diff
@@ -0,0 +1,173 @@
+                Interest.builder().id(1).department(Department.SYSC).keyword("testing").build(),
+                Interest.builder().id(2).department(Department.SYSC).keyword("machine_learning").build(),
+                Interest.builder().id(3).department(Department.MAAE).keyword("gears").build(),
+                Interest.builder().id(4).department(Department.SREE).keyword("grilled_cheese").build());
+
+        when(repo.findAll()).thenReturn(interests);
+        when(repo.findByDepartment(Department.SYSC)).thenReturn(interests.subList(0, 3));
+        when(repo.findByDepartment(Department.MAAE)).thenReturn(Collections.singletonList(interests.get(3)));
+        when(repo.findByDepartment(Department.SREE)).thenReturn(Collections.singletonList(interests.get(4)));
+        for(int i = 0; i <= 4; i++) {
+            when(repo.findById((long) i)).thenReturn(Optional.of(interests.get(i)));
+            doNothing().when(repo).deleteById((long) i);
+        }
+    }
+
+    /**
+     * helper method to convert objects into JSON format
+     *
+     * @param o the object to convert to JSON format
+     * @return a string contain the JSON for Object o
+     * @throws JsonProcessingException when JSON writing fails
+     */
+    private String toJson(Object o) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
+        ObjectWriter ow = mapper.writer().withDefaultPrettyPrinter();
+        return ow.writeValueAsString(o);
+    }
+
+    /**Test the retrieval of all interests*/
+    @Test
+    public void testGetAllInterests() throws Exception {
+        this.mockMvc.perform(get("/interest/"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(toJson(interests)));
+    }
+
+    /**Test the retrieval of specific interests by id*/
+    @Test
+    public void testGetInterestById() throws Exception {
+        for(Interest interest : interests){
+            this.mockMvc.perform(get("/interest/?id="+interest.getId()))
+                    .andExpect(status().isOk())
+                    .andExpect(content().json(toJson(interest)));
+        }
+    }
+
+    /**Test the retrieval of all interests listed under a particular department*/
+    @Test
+    public void testGetInterestByDepartment() throws Exception {
+        this.mockMvc.perform(get("/interest/department?department=SYSC"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(toJson(interests.subList(0, 3))));
+    }
+
+    /**Test the retrieval of the create interest page*/
+    @Test
+    public void testGetCreateInterestForm() throws Exception {
+        MvcResult result = this.mockMvc.perform(get("/interest/create"))
+                .andExpect(status().isOk()).andReturn();
+        assertTrue(result.getResponse().getContentAsString().contains("<title>Create Interest</title>"));
```

Comment:
Tests that do this aren't maintainable, but like a few other things in this PR, I don't know a better way with this current project setup.

No changes required for this PR